diff --git a/vllm/model_executor/layers/kda.py b/vllm/model_executor/layers/kda.py
index 70c67f33f0..8d1741176f 100644
--- a/vllm/model_executor/layers/kda.py
+++ b/vllm/model_executor/layers/kda.py
@@ -98,9 +98,18 @@ class KimiDeltaAttention(nn.Module, MambaBase):
 
     def get_state_shape(
         self,
+    ) -> tuple[tuple[int, ...], tuple[int, ...], tuple[int, ...], tuple[int, ...]]:
+        return self.get_state_shape_with_num_spec(0)
+
+    def get_state_shape_with_num_spec(
+        self, num_spec: int
     ) -> tuple[tuple[int, ...], tuple[int, ...], tuple[int, ...], tuple[int, ...]]:
         return MambaStateShapeCalculator.kda_state_shape(
-            self.tp_size, self.num_heads, self.head_dim, conv_kernel_size=self.conv_size
+            self.tp_size,
+            self.num_heads,
+            self.head_dim,
+            conv_kernel_size=self.conv_size,
+            num_spec=num_spec,
         )
 
     def __init__(
