diff --git a/vllm_ascend/patch/worker/patch_gdn_attn.py b/vllm_ascend/patch/worker/patch_gdn_attn.py
index 1ae910329..4eeaf89c8 100644
--- a/vllm_ascend/patch/worker/patch_gdn_attn.py
+++ b/vllm_ascend/patch/worker/patch_gdn_attn.py
@@ -740,7 +740,7 @@ def _patched_build(
         and attn_metadata.num_spec_decodes == 0
         and attn_metadata.num_decodes <= self.decode_cudagraph_max_bs
     ):
-        self.non_spec_state_indices_tensor[attn_metadata.num_actual_tokens :].fill_(NULL_BLOCK_ID)
+        self.non_spec_state_indices_tensor[common_attn_metadata.num_reqs :].fill_(NULL_BLOCK_ID)
     return attn_metadata
 
 
